zfit/core/basepdf.py

```diff
@@ -0,0 +1,1097 @@
+"""
+This  module defines the `BasePdf` that can be used to inherit from in order to build a custom PDF.
+
+The `BasePDF` implements already a lot of ready-to-use functionality like integral, automatic normalization
+and sampling.
+
+Defining your own pdf
+---------------------
+
+A simple example:
+>>> class MyGauss(BasePDF):
+>>>     def __init__(self, mean, stddev, name="MyGauss"):
+>>>         super().__init__(mean=mean, stddev=stddev, name=name)
+>>>
+>>>     def _unnormalized_prob(self, x):
+>>>         return tf.exp((x - mean) ** 2 / (2 * stddev**2))
+
+Notice that we only specify the *function* and no normalization (we could but would not change anything).
```

Comment:
I am not sure this sentence is clear. It sort of implies that it is not possible to specify integration?

Reply:
I agree. I extended it and also changed the formulation